### ISSUE TITLE
Destroy does not reset element to its origins

### ISSUE THREAD

**Issue by krnlde** [ariutta/svg-pan-zoom#171]
I have the case where I initialize and destroy an SVG Element based on its visibility and experienced, that the wrapping `g` with its `transform="matrix(...);"` attribute is not reset after destroying it. When then re-initializing the object, it will show the SVG at a veeery small scale as a starting point. Because the translation is then applied twice.

I tried to just remove the `transform` attribute by hand but that didn't produce my wanted result.

Any suggestions?

**Comment by bumbu** [comment]
One reason why transform attribute is not removed is because while initializing svgPanZoom it removes `viewBox` attribute. So removing transform would affect those SVGs that had `viewBox`. 

But from what I remember, destroying and reinitializing the same SVG was working fine. Can you please [provide an example](https://github.com/ariutta/svg-pan-zoom#bugs-and-issues) that doesn't work for you?

**Comment by krnlde** [comment]
https://jsfiddle.net/v0ym8gaL/

Howto: Initialize, Destroy, Initialize

**Comment by bumbu** [comment]
Removing `viewBox` attribute and setting `{fit: false, center: false}` [does not alter initial SVG state](https://jsfiddle.net/v0ym8gaL/1/).
In case when there is a `viewBox` attribute - it changes the zoom/pan on second initialization. 

I'll take a look and think about this when I'll have time. If you have a working solution then a pull-request will be welcome.

**Comment by krnlde** [comment]
No working solution right now. Maybe saving the viewBox on initialize is a solution? The other way would be to not use a matrix for zooming and panning, but the viewBox itself. But then the initial problem remains - saving the original viewBox for a later post-destroy-rollback.

**Comment by ttupper92618** [comment]
One point to consider is that ViewBox is not a given, so any code which assumes its existence will break under many use cases.

Sent from my iPhone

On Dec 10, 2015, at 3:38 AM, Kai Dorschner <notifications@github.com<mailto:notifications@github.com>> wrote:


No working solution right now. Maybe saving the viewBox on initialize is a solution? The other way would be to not use a matrix for zooming and panning, but the viewBox itself. But then the initial problem remains - saving the original viewBox for a later post-destroy-rollback.

-

**Comment by ttupper92618** [comment]
In your fiddle, you are not feeding in any options on your initialize.  By default, fit=true, as is center.  As bumbu points out above, If you modify your initialize to disable center and fit, it works as expected:

window.initialize = function () {
	instance = svgPanZoom('#demo-tiger', {
  	fit:false,
        center: false
  });
}

Am I misunderstanding your problem state?

**Comment by krnlde** [comment]
Having fit and center set to false works as expected. But this is considered a workaround, right?

**Comment by bumbu** [comment]
Yes, it is sort of workaround. On destroy we use `reset` method. The problem is that reset is based not on original SVG values, but on precomputed values based on `fit` and `center` values.
So we'd need to store SVG position and zoom before `fit` and `center` are applied and use these values on destroy.

**Comment by Anticom** [comment]
I wanted to use your package for an in-browser svg editor which get's serialized via innerHTML again. However destroying the pan-zoom doesn't restore the original state. I'd love to see that fixed so don't have to do a workarround for this use-case as well.
